src/sql/run/resultset.py

```diff
@@ -504,10 +505,11 @@ def _nonbreaking_spaces(match_obj):
 
 
 def _statement_is_select(statement):
-    statement_ = statement.lower().strip()
-    # duckdb also allows FROM without SELECT
-    return (
-        statement_.startswith("select")
-        or statement_.startswith("from")
-        or statement_.startswith("with")
-    )
+    if statement and isinstance(statement, str):
+        statement_ = statement.lower().strip()
+        # duckdb also allows FROM without SELECT
+        return sqlparse.parse(statement_)[
+            0
+        ].get_type() == "SELECT" or statement_.startswith("from")
```

Comment:
are there any cases where this would fail? e.g., sqlparse.parse will trigger an exception or return an empty list, causing `[0]` to fail or `[0]` to return something that doesn't have `get_type()` try with a bunch of things and see if it breaks

Reply:
list of things that can break it
1. passing empty value sqlparse.parse("")[0], we are handling it [here](https://github.com/ploomber/jupysql/pull/797/commits/2cfaac6230cae0cffead4ee3a6186d04a079fdad#diff-ee061806ed7a4f5a79d8571f64241031f5c61240dbe59e548de0666a7996ab88R508)
2. Anything except str input sqlparse.parse(4)[0]
3. This throws unknown, sqlparse.parse("FROM a")[0].get_type() but we handle it [here](https://github.com/ploomber/jupysql/pull/797/files#diff-ee061806ed7a4f5a79d8571f64241031f5c61240dbe59e548de0666a7996ab88R512)
4. This one throws unknown and is an active issue sqlparse.parse('(select 1 as "a") UNION (select 2 as "a")')[0].get_type()
5. comment also throws an unknown: sqlparse.parse('# select')[0].get_type()

Reply:
hi @edublancas let me know what you think